gp/individual.py

```diff
@@ -149,33 +129,14 @@ def to_sympy(self, simplify=True):
         -------
         SymPy expression
         """
-        return CartesianGraph(self.genome).to_sympy(simplify, self.parameter_names_to_values)
+        return CartesianGraph(self.genome).to_sympy(simplify)
 
     def update_parameters_from_torch_class(self, torch_cls):
-        """Update values stored in Parameter nodes of graph from parameters of
-        a given Torch instance.  Can be used to import new values from
-        a Torch class after they have been altered, e.g., by local
-        search.
+        self.genome.update_parameters_from_torch_class(torch_cls)
 
-        Parameters
-        ----------
-        torch_cls : torch.nn.module
-            Instance of a torch class.
-
-        Returns
-        -------
-        None
-
-        """
-        for name, value in torch_cls.named_parameters():
-            name = f"<{name[1:]}>"
-            if name in self.parameter_names_to_values:
-                self.parameter_names_to_values[name] = value.item()
-                assert not np.isnan(self.parameter_names_to_values[name])
-
-                # since a parameter of this individual has changed its
-                # fitness needs to be reevaluated
-                self.fitness = None
+        # since a parameter of this individual has changed its
+        # fitness needs to be reevaluated
+        self.fitness = None
```

Comment:
this is probably suboptimal since fitness only needs to be reevaluated when the value of a parameter which is used in the computational graph has changed, will look into this

Reply:
fixed